api4/remote_cluster.go

```diff
@@ -106,10 +108,18 @@ func remoteClusterAcceptMessage(c *Context, w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	// pass message to Remote Cluster Service
-	service.ReceiveIncomingMsg(rc, frame.Msg)
+	// pass message to Remote Cluster Service and write response
+	resp := service.ReceiveIncomingMsg(rc, frame.Msg)
+	if _, ok := resp[model.STATUS]; !ok {
```

Comment:
From the [ReceivedIncomingMsg](https://github.com/wiggin77/mattermost-server/blob/MM-28519_share_service/services/remotecluster/recv.go#L16) we do always return a status in the cases within the loop, the only case that we can return a response without a status is when we do not have any `listeners`. We can update `ReceivedIncomingMsg` to initialize `Response` with `statusOK` and eliminate this handling here.

Something like:

```go
func (rcs *Service) ReceiveIncomingMsg(rc *model.RemoteCluster, msg model.RemoteClusterMsg) Response {
	rcs.mux.RLock()
	defer rcs.mux.RUnlock()

	rcSanitized := *rc
	rcSanitized.Token = ""
	rcSanitized.RemoteToken = ""

	response := make(Response)
        response[ResponseStatusKey] = ResponseStatusOK

	listeners := rcs.getTopicListeners(msg.Topic)

	for _, l := range listeners {
		if err := callback(l, msg, &rcSanitized, response); err != nil {
			rcs.server.GetLogger().Log(mlog.LvlRemoteClusterServiceError, "Error from remote cluster message listener",
				mlog.String("msgId", msg.Id), mlog.String("topic", msg.Topic), mlog.String("remote", rc.DisplayName), mlog.Err(err))

			response[ResponseStatusKey] = ResponseStatusFail
			response[ResponseErrorKey] = err.Error()
		}
	}
	return response
}
```

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16638/commits/cfe00799ae130542f7eaadc54df79a8d9f5e6bff